polygeist/lib/Dialect/Polygeist/Transforms/SYCLRaiseHost.cpp

```diff
@@ -50,6 +51,99 @@ class SYCLRaiseHostConstructsPass
 // Pattern
 //===----------------------------------------------------------------------===//
 
+/// Searches the component of the split demangled function name containing the
+/// actual function name in \p components. Returns whether the actual function
+/// name is equal to \p functionName.
+static bool matchFunctionName(ArrayRef<StringRef> components,
+                              StringRef functionName) {
+  // First search the component containing the function name
+  std::size_t parenPos = 0;
+  components = components.drop_while([&](StringRef component) {
+    parenPos = component.find('(');
+    return parenPos == StringRef::npos;
+  });
+
+  // Not found
+  if (components.empty())
+    return false;
+
+  std::size_t nameLength = functionName.size();
+  if (parenPos < nameLength) {
+    // Component got is too short
+    return false;
+  }
+
+  StringRef component = components.front();
+  return functionName == component.substr(parenPos - nameLength, nameLength);
+}
+
+/// Returns whether the demangled function name \p functionName comes from a
+/// call to function \p name, member of \p typeName in namespace \p
+/// namespaceprefix.
+static bool checkFunction(StringRef functionName, StringRef namespacePrefix,
```

Comment:
Should we maybe rely on the Demangler for some of the functionality here? 
Concretely, the following functions might come in handy: 
* [`getFunctionDeclContextName`](https://llvm.org/doxygen/structllvm_1_1ItaniumPartialDemangler.html#aea51879b3b2822d1c0f74c71cff9bc2e)
* [`getFunctionBaseName`](https://llvm.org/doxygen/structllvm_1_1ItaniumPartialDemangler.html#a2effd15853bfffbeec4d08451f1fa81c)

Reply:
Done